integrationTests/singleShard/block/consensusNotAchieved_test.go

```diff
@@ -0,0 +1,143 @@
+package block
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConsensus_BlockWithoutTwoThirdsPlusOneSignaturesShouldNotBeAccepted(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	maxShards := 1
+	consensusGroupSize := 2
+	nodesPerShard := 5
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+
+	// create map of shards - testNodeProcessors for metachain and shard chain
+	nodesMap := integrationTests.CreateNodesWithNodesCoordinator(
+		nodesPerShard,
+		nodesPerShard,
+		maxShards,
+		consensusGroupSize,
+		consensusGroupSize,
+		integrationTests.GetConnectableAddress(advertiser),
+	)
+
+	for _, nodes := range nodesMap {
+		integrationTests.DisplayAndStartNodes(nodes)
+		integrationTests.SetEconomicsParameters(nodes, integrationTests.MaxGasLimitPerBlock, integrationTests.MinTxGasPrice, integrationTests.MinTxGasLimit)
+		//set rewards = 0 so we can easily test the balances taking into account only the tx fee
+		for _, n := range nodes {
+			n.EconomicsData.SetRewards(big.NewInt(0))
+		}
+	}
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, nodes := range nodesMap {
+			for _, n := range nodes {
+				_ = n.Node.Stop()
+			}
+		}
+	}()
+
+	fmt.Println("Delaying for nodes p2p bootstrap...")
+	time.Sleep(stepDelay)
+
+	round := uint64(0)
+	nonce := uint64(0)
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+
+	bitMapNotEnough := []byte{1}
+	body, hdr, _ := proposeBlock(nodesMap[0][0], round, nonce, bitMapNotEnough)
+	assert.NotNil(t, body)
+	assert.NotNil(t, hdr)
+
+	nodesMap[0][0].BroadcastBlock(body, hdr)
+	time.Sleep(stepDelay)
+
+	// the block should have not pass the interceptor
+	assert.Equal(t, int32(0), nodesMap[0][1].CounterHdrRecv)
+
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+
+	bitMapEnough := []byte{1, 0, 1, 0, 1}
+	body, hdr, _ = proposeBlock(nodesMap[0][0], round, nonce, bitMapEnough)
+	assert.NotNil(t, body)
+	assert.NotNil(t, hdr)
+
+	nodesMap[0][0].BroadcastBlock(body, hdr)
+	time.Sleep(stepDelay)
+
+	// this block should have passed the interceptor
+	assert.Equal(t, int32(1), nodesMap[0][1].CounterHdrRecv)
```

Comment:
If this block passed, then this is suspicious as the bitmap was not correctly constructed.

Reply:
it passed because the size of the bitmap was not checked. and the function only counted the number of ones in the bitmap.
Now I added a check for the bitmap size. Also, changed the logic for calculating the size when the bitmap is created.